### ISSUE TITLE
`import-x/extensions` rule doesn't work with Node.js subpath imports

### ISSUE THREAD

**Issue by sairus2k** [un-ts/eslint-plugin-import-x#437]
The import-x/extensions rule fails to check file extensions for [Node.js subpath imports](https://nodejs.org/api/packages.html#subpath-imports).

**Current Behavior**

When using subpath imports defined in `package.json`:

```json
{
  "imports": {
    "#utils/*": "./src/utils/*.js"
  }
}
```

The `import-x/extensions` rule doesn't validate extensions:

```typescript
import { helper } from '#utils/helper';     // ✓ No error (but should check extension)
import { helper } from '#utils/helper.js';  // Rule doesn't verify this either
```

**Expected Behavior**

The rule should check file extensions for subpath imports the same way it does for regular imports:

```typescript
// With "import-x/extensions": ["error", "always"]
import { helper } from '#utils/helper';     // ❌ Should error: missing extension
import { helper } from '#utils/helper.js';  // ✓ Should pass
```

**Root Cause**

The `parsePath` function treats `#` as a URL hash fragment, causing the pathname to be extracted incorrectly:

```typescript
export const parsePath = (path: string): ParsedPath => {
  const hashIndex = path.indexOf('#')
  const queryIndex = path.indexOf('?')
  const hasHash = hashIndex !== -1
  const hash = hasHash ? path.slice(hashIndex) : ''
  const pathname = hasHash ? path.slice(0, hashIndex) : path
  // For '#utils/helper', this results in pathname: '' and hash: '#utils/helper'
}
```

This causes the `import-x/extensions` rule (and potentially other rules) to skip validation since it doesn't see a valid pathname.

**Suggested Fix**

The `parsePath` function should not treat URLs started with the symbol `#` as hash fragments.

```typescript
export const parsePath = (path: string): ParsedPath => {
  const hashIndex = path.indexOf('#')
  const hasHash = hashIndex > 0
  // ... rest of the existing logic
}
```

**Environment**
- Node.js version: v24.11.0
- eslint-plugin-import-x version: v4.16.1 (latest at the moment)

**Additional Context**

Resolution appears to work correctly, but path parsing for rule validation is broken.

Other rules that depend on `parsePath` might also be affected.
